ios/Classes/TiNfcNfcAdapterProxy.m

```diff
@@ -105,13 +186,61 @@ - (void)readerSession:(NFCNDEFReaderSession *)session didInvalidateWithError:(NS
         [_nNdefInvalidated call:@[ @{
           @"cancelled" : @(error.code == 200),
           @"message" : [error localizedDescription],
-          @"code" : NUMINTEGER([error code])
+          @"code" : NUMINTEGER([error code]),
+          @"type" : NDEF_READER_SESSION
+        } ]
+                     thisObject:self];
+      },
+      NO);
+}
+
+#pragma mark NFCReaderSessionDelegate
+
+- (void)tagReaderSession:(NFCTagReaderSession *)session didInvalidateWithError:(NSError *)error
+{
+  _nfcTagReadersession = nil;
+
+  if (!_nNdefInvalidated) {
+    return;
+  }
+  TiThreadPerformOnMainThread(
+      ^{
+        [_nNdefInvalidated call:@[ @{
+          @"cancelled" : @(error.code == 200),
+          @"message" : [error localizedDescription],
+          @"code" : NUMINTEGER([error code]),
+          @"type" : NFC_TAG_READER_SESSION
         } ]
                      thisObject:self];
       },
       NO);
 }
 
+- (void)tagReaderSessionDidBecomeActive:(NFCTagReaderSession *)session
+{
+
+  if (![self _hasListeners:@"tagReaderSessionDidBecomeActive"]) {
+    return;
+  }
+  [self fireEvent:@"tagReaderSessionDidBecomeActive"
+       withObject:@{
+         @"type" : NFC_TAG_READER_SESSION
+       }];
+  [self fireEvent:@"tagReaderSessionDidBecomeActive"];
```

Comment:
@anil-shukla-axway , Why we are calling ```tagReaderSessionDidBecomeActive``` twice.

Reply:
Removed